### PR TITLE
Add skeleton for JAVA API

### PR DESCRIPTION
**Pull request by apraovjr**
**Purpose**
- Implement JAVA API for POST AND PATCH userprofile 
- First iteration is just basic skeleton of controller, models
- Healthcheck route api works

### CODE REVIEW

File: apis/user-java/src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+springfox.documentation.swagger.v2.path=/api-docs
```
Comment (dtzar) [inline apis/user-java/src/main/resources/application.properties]: docs base path should be `/api/docs/userjava`

**Comment by apraovjr** [comment]
@dtzar why does the `profile` needs to be removed? Isn't it the model class we need for the user

File: apis/user-java/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+This server was generated by the [swagger-codegen](https://github.com/swagger-api/swagger-codegen) project.  
+By using the [OpenAPI-Spec](https://github.com/swagger-api/swagger-core), you can easily generate a server stub.  
+This is an example of building a swagger-enabled server in Java using the SpringBoot framework.  
+
+The underlying library integrating swagger to SpringBoot is [springfox](https://github.com/springfox/springfox)  
+
+Start your server as an simple java application  
+
+You can view the api documentation in swagger-ui by pointing to  
+http://localhost:8080/
```
Comment (dtzar) [inline apis/user-java/README.md]: Can you please add the current URLS which actually serve the dump of the swagger JSON and the one which serves the html page?

File: apis/user-java/src/main/resources/application.properties (hunk 1 above)
Comment (dtzar) [reply apis/user-java/src/main/resources/application.properties]: As discussed this, breaks things when changed and will need more digging - so shelving for now.

**Pull request merged**